crates/consensus/src/engine/mod.rs

```diff
@@ -194,9 +204,29 @@ impl<Client: HeaderProvider + BlockProvider> ConsensusEngine for EthConsensusEng
             })
         }
 
-        // TODO: execute block
-
-        Ok(PayloadStatus::new(PayloadStatusEnum::Valid, block.hash()))
+        let (header, body, _) = block.split();
+        let transactions = body
+            .into_par_iter()
```

Comment:
I don't think we should use rayon par_iter here for two reasons:

* the overhead for spinning up a threadpool for transactions of a single block may be larger, especially on less powerful machines that also run the rest of the node
* this could compete with tokio's threadpool for resources

I'd recommend we do this sequentially and try a benchmark with 50,100,200 tx in a block

Reply:
this makes sense to me. let's table par iter usage until we do benchmarks then. opened #641 to track